src/types/app_config.go

```diff
@@ -125,3 +126,18 @@ func (a AppConfig) forEachService(fn func(string, string, ServiceData)) {
 		fn("public", serviceName, data)
 	}
 }
+
+func (a AppConfig) validateAppConfig() error {
+	appNameRegex := regexp.MustCompile("^[a-z0-9]+(-[a-z0-9]+)*$")
+	if !appNameRegex.MatchString(a.Name) {
+		return errors.New("only lowercase alphanumeric character(s) separated by a single hyphen are allowed. Must match regex: /^[a-z0-9]+(-[a-z0-9]+)*$/")
```

Comment:
Can this error message include the fact that error occurs in `application.yaml`, that its the `name` field, and the offending value?

Maybe something like. `The "name" field ("invalid-app") in "application.yml" is invalid. ...`